tests/integration_test.rs

```diff
@@ -0,0 +1,133 @@
+extern crate addr2line;
+extern crate itertools;
+extern crate glob;
+
+use std::env;
+use std::path;
+use std::process;
+use std::io::prelude::*;
+use itertools::Itertools;
+
+#[test]
+fn identity_map() {
+    let target = env::current_exe().unwrap();
+    let mut debug = target.clone();
+
+    // point us at the real macOS debugging files
+    if cfg!(target_os = "macos") {
+        let fname = debug.file_name().unwrap().to_string_lossy().into_owned();
+        debug.set_file_name(format!("{}.dSYM", fname));
+        debug.push("Contents");
+        debug.push("Resources");
+        debug.push("DWARF");
+        debug.push(fname);
+    }
+
+    // create our mapping
+    let ours = addr2line::Mapping::new(&debug).unwrap();
+
+    // spin up the "real" addr2line
+    let (theirs, exe) = if cfg!(target_os = "macos") {
+        ("atos", "-o")
+    } else {
+        ("addr2line", "-e")
+    };
+    let mut theirs = process::Command::new(theirs)
+        .arg(exe)
+        .arg(&target)
+        .stdin(process::Stdio::piped())
+        .stdout(process::Stdio::piped())
+        .spawn()
+        .expect("failed to spawn their addr2line");
+
+    // find a bunch of addresses
+    let names = process::Command::new("/usr/bin/nm")
+        .arg(&target)
+        .output()
+        .expect("failed to execute nm");
+    let mappings = String::from_utf8_lossy(&names.stdout);
+
+    // ideally, we'd just loop over mappings and do them one by one.
+    // unfortunately, atos on macOS doesn't flush its output for every
+    // input line, and so this doesn't work. instead, we pass all the
+    // mappings to `theirs`, close stdin, and then start walking.
+    let mut theirs_in = theirs.stdin.take().unwrap();
+    let mappings: Vec<_> = mappings.lines()
+        .filter_map(|map| {
+            let mut fields = map.split_whitespace();
+            let addr = fields.next().expect("nm address missing");
+            let t = fields.next().unwrap();
+            if t != "T" && t != "t" {
+                // we only want functions, which are in the text section
+                None
+            } else {
+                Some(addr)
+            }
+        })
+        .step(5)
+        .map(|addr| {
+            theirs_in.write_all("0x".as_bytes()).unwrap();
+            theirs_in.write_all(addr.as_bytes()).unwrap();
+            theirs_in.write_all("\n".as_bytes()).unwrap();
+            u64::from_str_radix(addr, 16).unwrap()
+        })
+        .collect();
+    drop(theirs_in);
+
+    // go through addresses one by one
+    use std::io::BufReader;
+    let mut theirs_out = BufReader::new(theirs.stdout.take().unwrap()).lines();
+    let mut all = 0;
+    let mut excusable = 0;
+    let mut got = 0;
+    for addr in mappings {
+        // read the oracle ouput
+        let theirs = theirs_out.next().unwrap().unwrap();
+
+        // parse it if necessary
+        let theirs = if cfg!(target_os = "macos") {
```

Comment:
Can you factor this parsing out into its own function that returns a tuple `(Option<String>, Option<usize>)` for filename and line number (or `None` when they are `?`).